classes/clean_moodle_url.php

```diff
@@ -286,6 +286,19 @@ public static function clean(\moodle_url $orig) {
                     self::log("Rewrite user profile");
                 }
             }
+
+            // Clean up user profile urls in forum posts
+            if ($path == "/mod/forum/user.php" && $params['id'] && $params['mode']) {
+                $slug = $DB->get_field('user', 'username', array('id' => $params['id']));
+                $slug = urlencode($slug);
+                $newpath = "/user/$slug";
+                if (!is_dir($CFG->dirroot . $newpath) && !is_file($CFG->dirroot . $newpath . ".php")) {
+                    $path = $newpath . '/' . $params['mode'];
```

Comment:
Related to my other comment, how many modes are their? eg mode == 'discussion', should they all be mapped? Should only discussions be mapped? If there are others then add unit tests for them too